sdk/eventhub/azure-eventhub/azure/eventhub/_common.py

```diff
@@ -525,8 +503,18 @@ class EventDataBatch(object):
      Event Hub decided by the service.
     """
 
-    def __init__(self, max_size_in_bytes=None, partition_id=None, partition_key=None):
-        # type: (Optional[int], Optional[str], Optional[Union[str, bytes]]) -> None
+    def __init__(
+        self,
+        max_size_in_bytes: Optional[int] = None,
+        partition_id: Optional[str] = None,
+        partition_key: Optional[Union[str, bytes]] = None,
+        **kwargs,
```

Comment:
@annatisch - adding kwargs is the only public API change (other than making the `EventData.message` a property instead of an ivar). B/c we recommend creating this object with producer_client.create_batch(), I assumed that we would be able to just pass the amqp transport to the EventDataBatch internally. However, since it's possible that users are constructing this manually, I was thinking that we could do the following in the future once we expose the `uamqp_transport` kwarg.

```
def __init__(
        self,
        max_size_in_bytes: Optional[int] = None,
        partition_id: Optional[str] = None,
        partition_key: Optional[Union[str, bytes]] = None,
        **kwargs,
):
try:
  self._amqp_transport = kwargs.pop("amqp_transport")
except KeyError:
  try:
    uamqp_transport = kwargs.pop("uamqp_transport")
    self._amqp_transport = UamqpTransport if uamqp_transport else PyamqpTransport
  except KeyError:
    raise TypeError("uamqp_transport is a required keyword.")
```

One other potential option would be to move out BatchMessage construction from here to in `send_batch`. However, I'm thinking that:
1) It might not be efficient to have to add the Messages to the BatchMessage in a loop in send instead of in add here?
2) The message sizes seem to be different b/w uamqp Message and pyamqp Message for some reason. So, when we check whether the batch is full, it might differ from when that error would occur for uamqp Message vs pyamqp Message.

Reply:
Should we wait to make this API change until pyamqp gets merged, so that we don't have to bump minor version number until then? If so, I can change this so that we don't take kwargs and have: ` self._amqp_transport = UamqpTransport`.

Reply:
Yes I would leave it out of the API for now and just default to UamqpTransport.
I think we can make the Batch Message itself transport agnostic (unless it's instantiated using `client.send_batch()`).
The only thing it needs the transport for is tracking the encoded size - and while, as you say, there's some difference in the exact byte count between the two implementations, I think we'd be pretty save using pyamqp encoder (which will encodes superfluous data and so is usually bigger).

Reply:
updated so that EventDataBatch doesn't take kwargs. Will make EventDataBatch transport agnostic in the pyamqp switch PR.